### DISCUSSION TITLE
phpunit 10 `@covers` annotation vs. `CoversClass` attribute and covering class methods

### DISCUSSION THREAD

**Discussion by mfn** [infection/infection#1905] [Q&A]
Hi,

I've a project I'm looking at to switch from phpunit `@covers` annotations to their new attributes, specifically `\PHPUnit\Framework\Attributes\CoversClass`.

However the attribute syntax does not support `ClassUnderTest::method`, whilst `@covers` did. I use this a lot to "contain" the mutation impact area.

After the switch I get a lot more escaped mutants and I think it's due to this change now.

Does infection support `CoversClass`? If yes, is there a solution? Would I need to use `CoversClass` _and_ `@covers` to still support the `::method` syntax.

Currently I'm not sure if it's my fault or not, I'm still figuring out things.

thx

**Comment by maks-rafalko** [comment]
Can I ask you to provide an example/commit e2e test/repo where it works woth annotation but doesn't work with an attribute? Personally I didn't use it so don't know if there is a bug

**Reply by mfn** [reply]
I'm not sure I found the time, didn't expect to be the first one :p

Found more evidence this won't come to phpunit, I found a PR which was rejected with the same argument of that granularity not being useful -> https://github.com/sebastianbergmann/phpunit/pull/5360
